lib/routes/routeBackbeat.js

```diff
@@ -650,6 +664,44 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
                     return next();
                 });
             },
+            async () => {
+                if (versioning && !objMd) {
+                    let masterMD;
+
+                    try {
+                        masterMD = await metadata.getObjectMDPromised(bucketName, objectKey, {}, log);
+                    } catch (err) {
+                        if (err.is?.NoSuchKey) {
+                            log.debug('no master found for versioned object', {
+                                method: 'putMetadata',
+                                bucketName,
+                                objectKey,
+                            });
+                        } else {
+                            throw err;
+                        }
+                    }
+
+                    if (!masterMD) {
+                        return;
+                    }
+
+                    const versioningPreprocessingResult =
+                        await versioningPreprocessingPromised(bucketName, bucketInfo, objectKey, masterMD, log);
+
+                    if (versioningPreprocessingResult?.nullVersionId) {
+                        omVal.nullVersionId = versioningPreprocessingResult.nullVersionId;
+                        omVal.versionId = versioningPreprocessingResult.versionId;
+                        omVal.versioning = versioningPreprocessingResult.versioning;
+                        omVal.isNull = versioningPreprocessingResult.isNull;
+                        omVal.deleteNullKey = versioningPreprocessingResult.deleteNullKey;
+
+                        if (options.extraMD) {
+                            Object.assign(omVal, versioningPreprocessingResult.extraMD);
+                        }
```

Comment:
nit: (not sure the `|| {}`) is even needed, but seems best to avoid the extra cognitive complexity?
```suggestion
                        Object.assign(omVal, versioningPreprocessingResult.extraMD || {});
```